Tools/autotest/arduplane.py

```diff
@@ -5529,6 +5531,26 @@ def MAV_CMD_GUIDED_CHANGE_ALTITUDE(self):
             timeout=30,
             relative=True,
         )
+        # test that this works if switching between RELATIVE (HOME) and GLOBAL(AMSL)
+        self.start_subtest("set global/AMSL altitude switch to loiter and resume")
+        alt = 600
+        self.run_cmd_int(
+            mavutil.mavlink.MAV_CMD_GUIDED_CHANGE_ALTITUDE,
+            p7=alt,
+            frame=mavutil.mavlink.MAV_FRAME_GLOBAL,
+        )
+        self.wait_altitude(alt-3, alt+3, timeout=30, relative=False)
+        self.change_mode('LOITER')
+        self.delay_sim_time(5)
```

Comment:
Shouldn't there be a `wait_altitude` in here?

Reply:
I've added an altitude_source = "TERRAIN_REPORT.current_height" to wait_altitude. It's a big clunky but it does work.